recon_surf/recon-surf.sh

```diff
@@ -1002,36 +1044,39 @@ for hemi in lh rh ; do
-  longmaxdist=""
-  if [ "$long" == "1" ] ; then
-    inputsurf="../surf/$hemi.orig_white"
-    longmaxdist="--max-cbv-dist 3.5"
+  cmd="mris_place_surface --adgws-in ../surf/autodet.gw.stats.${hemi}.dat --seg aseg.presurf.mgz \
+    --threads $threads --wm $wm_file --invol brain.finalsurfs.mgz --$hemi --o ../surf/${hemi}.white \
+    --white --nsmooth 0 --rip-label ../label/${hemi}.cortex.label \
+    --rip-bg --rip-surf ../surf/${hemi}.white.preaparc --aparc $aparc"
+  if [ "$long" == "0" ] ; then # cross/regular/base
+    cmd="$cmd --i ../surf/$hemi.white.preaparc"
+  else  # longitudinal processing ; also adds longmaxdist
+    cmd="$cmd --i ../surf/$hemi.orig_white --max-cbv-dist 3.5"
   fi
-  cmd="mris_place_surface --adgws-in ../surf/autodet.gw.stats.${hemi}.dat --seg aseg.presurf.mgz --threads $threads --wm wm.mgz --invol brain.finalsurfs.mgz --$hemi --i $inputsurf --o ../surf/${hemi}.white --white --nsmooth 0 --rip-label ../label/${hemi}.cortex.label --rip-bg --rip-surf ../surf/${hemi}.white.preaparc --aparc $aparc $longmaxdist"
   RunIt "$cmd" "$LF" "$CMDF"
 
   # CREAT PIAL SURFACE
   # 4 min compute pial :
-  inputsurf="../surf/$hemi.white"
-  longmaxdist=""
-  if [ "$long" == "1" ] ; then
-    inputsurf="../surf/$hemi.orig_pial"
-    longmaxdist="--max-cbv-dist 3.5 --blend-surf .25 ../surf/$hemi.white"
+  cmd="mris_place_surface --adgws-in ../surf/autodet.gw.stats.${hemi}.dat --seg aseg.presurf.mgz \
+    --threads $threads --wm $wm_file --invol brain.finalsurfs.mgz --$hemi --o ../surf/${hemi}.pial.T1 \
+    --pial --nsmooth 0 --rip-label ../label/${hemi}.cortex+hipamyg.label \
+    --pin-medial-wall ../label/${hemi}.cortex.label --aparc $aparc \
+    --repulse-surf ../surf/${hemi}.white --white-surf ../surf/${hemi}.white"
+  if [ "$long" == "0" ] ; then # cross/regular/base
+    cmd="$cmd --i ../surf/$hemi.white"
+  else  # longitudinal processing ; also adds longmaxdist
+    cmd="$cmd --i ../surf/$hemi.orig_pial --max-cbv-dist 3.5 --blend-surf .25 ../surf/$hemi.white"
   fi
-  cmd="mris_place_surface --adgws-in ../surf/autodet.gw.stats.${hemi}.dat --seg aseg.presurf.mgz --threads $threads --wm wm.mgz --invol brain.finalsurfs.mgz --$hemi --i $inputsurf --o ../surf/${hemi}.pial.T1 --pial --nsmooth 0 --rip-label ../label/${hemi}.cortex+hipamyg.label --pin-medial-wall ../label/${hemi}.cortex.label --aparc $aparc --repulse-surf ../surf/${hemi}.white --white-surf ../surf/${hemi}.white $longmaxdist"
   RunIt "$cmd" "$LF" "$CMDF"
 
   echo "popd > /dev/null" >> "$CMDF"
 
   # Here insert DoT2Pial  later --> if T2pial is not run, need to softlink pial.T1 to pial!
   echo "pushd $sdir > /dev/null" >> "$CMDF"
   softlink_or_copy "$hemi.pial.T1" "$hemi.pial" "$LF" "$CMDF"
-  echo "popd > /dev/null" >> "$CMDF"
 
   # these are run automatically in fs7* recon-all and cannot be called directly without -pial flag (or other t2 flags)
   # they are the same for fsaparc and long
-  echo "pushd $mdir > /dev/null" >> "$CMDF"
```

Comment:
Have you tested this? I am not sure why it was cd into mri dir, but maybe there was a reason. With your change it is now running from within surf dir which might change things.

Reply:
I had tested some of these, but this clearly is a bug :-(